core/ops_builtin.rs

```diff
@@ -411,3 +417,131 @@ fn op_is_terminal(
   let handle = state.resource_table.get_handle(rid)?;
   Ok(handle.is_terminal())
 }
+
+async fn do_load_job<'s>(
+  scope: &mut v8::HandleScope<'s>,
+  module_map_rc: Rc<ModuleMap>,
+  specifier: &str,
+  code: Option<String>,
+) -> Result<ModuleId, Error> {
+  if let Some(code) = code {
+    module_map_rc
+      .new_es_module(scope, false, specifier.to_owned(), code, false, None)
+      .map_err(|e| e.into_any_error(scope, false, false))?;
+  }
+
+  let mut load = ModuleMap::load_side(module_map_rc.clone(), specifier).await?;
+
+  while let Some(load_result) = load.next().await {
+    let (request, info) = load_result?;
+    load
+      .register_and_recurse(scope, &request, info)
+      .map_err(|e| e.into_any_error(scope, false, false))?;
+  }
+
+  let root_id = load.root_module_id.expect("Root module should be loaded");
+  module_map_rc
+    .instantiate_module(scope, root_id)
+    .map_err(|e| {
+      let exception = v8::Local::new(scope, e);
+      exception_to_err_result::<()>(scope, exception, false, false).unwrap_err()
+    })?;
+
+  Ok(root_id)
+}
+
+/// Wrap module with another module that also exports `__esModule=true`.
+fn wrap_module<'s>(
+  scope: &mut v8::HandleScope<'s>,
+  module: v8::Local<'s, v8::Module>,
+) -> Option<v8::Local<'s, v8::Module>> {
+  const SOURCE: &str = "
+  export * from 'original';
+  export {default} from 'original';
+  export const __esModule = true;";
+
+  let source = v8::String::new(scope, SOURCE)?;
+  let origin = v8::ScriptOrigin::new(
+    scope,
+    source.into(),
+    0,
+    0,
+    false,
+    0,
+    None,
+    true,
+    false,
+    true,
+    None,
+  );
+
+  let mut source = v8::script_compiler::Source::new(source, Some(&origin));
+  let wrapper_module = v8::script_compiler::compile_module(scope, &mut source)?;
+
+  let global_module = v8::Global::new(scope, module);
+  scope.set_slot(global_module);
+  fn resolve_callback<'s>(
+    context: v8::Local<'s, v8::Context>,
+    _: v8::Local<'s, v8::String>,
+    _: v8::Local<'s, v8::FixedArray>,
+    _: v8::Local<'s, v8::Module>,
+  ) -> Option<v8::Local<'s, v8::Module>> {
+    // SAFETY: It is safe to open a CallbackScope from a context in this callback.
+    let mut scope = unsafe { v8::CallbackScope::new(context) };
+    let module = scope.get_slot::<v8::Global<v8::Module>>().unwrap().clone();
+    scope.remove_slot::<v8::Global<v8::Module>>();
+    Some(v8::Local::new(&mut scope, module))
+  }
+  wrapper_module.instantiate_module(scope, resolve_callback)?;
+
+  wrapper_module.evaluate(scope)?;
+
+  Some(module)
+}
+
+#[op2(reentrant)]
+fn op_import_sync<'s>(
+  scope: &mut v8::HandleScope<'s>,
+  #[string] specifier: &str,
+  #[string] code: Option<String>,
+) -> Result<v8::Local<'s, v8::Value>, Error> {
+  let module_map_rc = JsRealm::module_map_from(scope);
+
+  // no js execution within block_on
+  let module_id = futures::executor::block_on(do_load_job(
```

Comment:
This might be problematic - in the past every time we did `block_on` outside of testing code it caused a bug in the end. But let's see if we can integrate it into Deno